File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    progresses = deque(progresses)
    speeds = deque(speeds)
    answer = list()
    
    while progresses:
        cnt = 0

        for i in range(len(progresses)):
            progresses[i] += speeds[i]

        while progresses:
            if progresses[0] >= 100:
                cnt += 1
                progresses.popleft()
                speeds.popleft()
            else:
                if cnt:
                    answer.append(cnt)
                break
        
        if not progresses and cnt:
            answer.append(cnt)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> time;

    for (int i = progresses.size() - 1; i >= 0; i--) {
        time.emplace_back(ceil((100.0 - progresses[i]) / speeds[i]));
    }
    
    while (!time.empty()) {
        int cnt = 1;
        int last = time.back();
        
        time.pop_back();
        
        while (!time.empty() && time.back() <= last) {
            cnt++;
            time.pop_back();
        }
        
        answer.emplace_back(cnt);
    }
    
    return answer;
}